src/screens/WalletScreen/index.tsx

```diff
@@ -48,8 +49,10 @@ function WalletScreen() {
   const walletReady = useSelector(({ appState: { walletReady } }: AppState) => walletReady);
   const { isWalletEthZero, isLoadingUserAssets, isLoadingBalance, briefSectionsData: walletBriefSectionsData } = useWalletSectionsData();
 
-  const trackWallets = useCallback(() => {
-    const identify = Object.values(wallets || {}).reduce(
+  useEffect(() => {
+    if (!wallets) return;
+
+    const identify = Object.values(wallets).reduce(
```

Comment:
i think we only want to call `trackWallets` after the wallet has been initialized as it was previously written. This will do it I think on mount instead

Reply:
Before, this function was called right after the initialize wallet function had been called, but that function executing offered no guarantee that the wallets variable the trackWallets uses was ready. In testing locally it was usually the case that wallets was still null when trackWallets was called. 

This instead uses the wallets variable as a dependency of the effect hook which is safer and also handles the case of calling track again when wallets are deleted or added.

Reply:
That makes sense. Thanks for the explanation!